src/PowerShellEditorServices.Hosting/EditorServicesRunner.cs

```diff
@@ -0,0 +1,257 @@
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Threading.Tasks;
+using SMA = System.Management.Automation;
+
+namespace Microsoft.PowerShell.EditorServices.Hosting
+{
+    /// <summary>
+    /// Class to manage the startup of PowerShell Editor Services.
+    /// This should be called by <see cref="EditorServicesLoader"/> only after Editor Services has been loaded.
+    /// </summary>
+    internal class EditorServicesRunner : IDisposable
+    {
+        /// <summary>
+        /// Create a new Editor Services runner.
+        /// </summary>
+        /// <param name="logger">The host logger to log through.</param>
+        /// <param name="config">The startup configuration to use.</param>
+        /// <param name="sessionFileWriter">The session file writer to use.</param>
+        /// <returns></returns>
+        public static EditorServicesRunner Create(
+            HostLogger logger,
+            EditorServicesConfig config,
+            ISessionFileWriter sessionFileWriter,
+            IReadOnlyCollection<IDisposable> loggersToUnsubscribe)
+        {
+            return new EditorServicesRunner(logger, config, sessionFileWriter, loggersToUnsubscribe);
+        }
+
+        private readonly HostLogger _logger;
+
+        private readonly EditorServicesConfig _config;
+
+        private readonly ISessionFileWriter _sessionFileWriter;
+
+        private readonly EditorServicesServerFactory _serverFactory;
+
+        private readonly IReadOnlyCollection<IDisposable> _loggersToUnsubscribe;
+
+        private bool _alreadySubscribedDebug;
+
+        private EditorServicesRunner(
+            HostLogger logger,
+            EditorServicesConfig config,
+            ISessionFileWriter sessionFileWriter,
+            IReadOnlyCollection<IDisposable> loggersToUnsubscribe)
+        {
+            _logger = logger;
+            _config = config;
+            _sessionFileWriter = sessionFileWriter;
+            _serverFactory = EditorServicesServerFactory.Create(_config.LogPath, (int)_config.LogLevel, logger);
+            _alreadySubscribedDebug = false;
+            _loggersToUnsubscribe = loggersToUnsubscribe;
+        }
+
+        /// <summary>
+        /// Start and run Editor Services and then wait for shutdown.
+        /// </summary>
+        /// <returns>A task that ends when Editor Services shuts down.</returns>
+        public async Task RunUntilShutdown()
+        {
+            // Start Editor Services
+            Task runAndAwaitShutdown = CreateEditorServicesAndRunUntilShutdown();
+
+            // Now write the session file
+            _logger.Log(PsesLogLevel.Diagnostic, "Writing session file");
+            _sessionFileWriter.WriteSessionStarted(_config.LanguageServiceTransport, _config.DebugServiceTransport);
+
+            // Finally, wait for Editor Services to shut down
+            await runAndAwaitShutdown.ConfigureAwait(false);
+        }
+
+        public void Dispose()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4579423)